pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -569,27 +568,46 @@ func getFirstContainerWithSourceVolume(containers []corev1.Container) (string, s
 }
 
 // Delete deletes the component
-func (a Adapter) Delete(labels map[string]string) error {
-	spinner := log.Spinnerf("Deleting devfile component %s", a.ComponentName)
-	defer spinner.End(false)
+func (a Adapter) Delete(labels map[string]string, show bool) error {
 
-	componentExists, err := utils.ComponentExists(a.Client, a.ComponentName)
+	log.Infof("\nGathering information for component %s", a.ComponentName)
+	podSpinner := log.Spinner("Checking status for component")
+	defer podSpinner.End(false)
+
+	pod, err := a.Client.GetPodUsingComponentName(a.ComponentName)
 	if kerrors.IsForbidden(err) {
 		klog.V(4).Infof("Resource for %s forbidden", a.ComponentName)
 		// log the error if it failed to determine if the component exists due to insufficient RBACs
-		spinner.End(false)
+		podSpinner.End(false)
 		log.Warningf("%v", err)
 		return nil
+	} else if e, ok := err.(*kclient.NoPodFoundError); ok {
```

Comment:
wouldn't this error(kclient.NoPodFoundError) be returned by kubernetes also? do we need to define a separate one?

Reply:
The kclient API uses a list API, so in case of 0, it returns an empty array. We had to change the kclient API being used with this PR from `utils.ComponentExists()` to `a.Client.GetPodUsingComponentName()`

here is what it does in kclient:
```
pods, err := c.KubeClient.CoreV1().Pods(c.Namespace).List(metav1.ListOptions{
		LabelSelector: selector,
	})
	if err != nil {
		// Dont wrap error since we want to know if its a forbidden error
		// if we wrap, we lose the err status reason and callers of this api rely on it
		return nil, err
	}
	numPods := len(pods.Items)
	if numPods == 0 {
		return nil, &NoPodFoundError{Selector: selector}
	} else if numPods > 1 {
		return nil, fmt.Errorf("multiple Pods exist for the selector: %v. Only one must be present", selector)
	}
```